torchbenchmark/models/torchrec_dlrm/__init__.py

```diff
@@ -0,0 +1,156 @@
+
+from .args import parse_args, InteractionType
+
+
+class Model(BenchmarkModel):
+    task = RECOMMENDATION.RECOMMENDATION
+    # upstream default batch size: 32
+    # https://github.com/facebookresearch/dlrm/blob/main/torchrec_dlrm/dlrm_main.py#L86
+    DEFAULT_TRAIN_BSIZE = 32
+    DEFAULT_EVAL_BSIZE = 32
+
+    def __init__(self, test, device, jit, batch_size=None, extra_args=[]):
+        super().__init__(test=test, device=device, jit=jit, batch_size=batch_size, extra_args=extra_args)
+        args = parse_args(self.extra_args)
+        backend = "nccl" if self.device == "cuda" else "gloo"
+        # initialize example data
+        if self.test == "train":
+            args.batch_size = self.batch_size
+            loader = get_dataloader(args, backend, "train")
+        if self.test == "eval":
+            args.test_batch_size = self.batch_size
+            loader = get_dataloader(args, backend, "test")
+        limit_batches = 1
+        iterator = itertools.islice(iter(loader), limit_batches)
+        self.example_inputs = next(iterator).to(self.device)
+        # parse the args
+        args.dense_arch_layer_sizes = [int(x) for x in args.dense_arch_layer_sizes.split(',') if x.strip().isdigit()]
+        args.over_arch_layer_sizes = [int(x) for x in args.over_arch_layer_sizes.split(',') if x.strip().isdigit()]
+        args.interaction_branch1_layer_sizes = [int(x) for x in args.interaction_branch1_layer_sizes.split(',') if x.strip().isdigit()]
+        args.interaction_branch2_layer_sizes = [int(x) for x in args.interaction_branch2_layer_sizes.split(',') if x.strip().isdigit()]
+
+        assert args.in_memory_binary_criteo_path == None and args.synthetic_multi_hot_criteo_path == None, \
+            f"Torchbench only supports random data inputs."
+
+        eb_configs = [
+            EmbeddingBagConfig(
+                name=f"t_{feature_name}",
+                embedding_dim=args.embedding_dim,
+                num_embeddings=none_throws(args.num_embeddings_per_feature)[feature_idx]
+                if args.num_embeddings is None
+                else args.num_embeddings,
+                feature_names=[feature_name],
+            )
+            for feature_idx, feature_name in enumerate(DEFAULT_CAT_NAMES)
+        ]
+        if args.interaction_type == InteractionType.ORIGINAL:
+            dlrm_model = DLRM(
+                embedding_bag_collection=EmbeddingBagCollection(
+                    tables=eb_configs, device=self.device
+                ),
+                dense_in_features=len(DEFAULT_INT_NAMES),
+                dense_arch_layer_sizes=args.dense_arch_layer_sizes,
+                over_arch_layer_sizes=args.over_arch_layer_sizes,
+                dense_device=device,
+            )
+        elif args.interaction_type == InteractionType.DCN:
+            dlrm_model = DLRM_DCN(
+                embedding_bag_collection=EmbeddingBagCollection(
+                    tables=eb_configs, device=self.device
```

Comment:
Originally device was `meta`, need to update it to `self.device`. Otherwise there's error like "RuntimeError: Tensor on device meta is not on the expected device cuda:0!"